lib/spack/spack/environment/environment.py

```diff
@@ -599,6 +689,16 @@ def regenerate(self, concretized_root_specs):
             tty.debug("View at %s does not need regeneration." % self.root)
             return
 
+        if update_method == "exchange" and os.path.isdir(new_root):
+            # If new_root is the newest thing in its directory, no need to update
+            parent = os.path.dirname(new_root)
+            siblings = [os.path.join(parent, s) for s in os.listdir(parent)]
+            siblings.sort(reverse=True, key=lambda p: os.stat(p).st_mtime)
+            if siblings[0] == new_root:
```

Comment:
```suggestion
            if max(siblings, key=...) == new_root:
```

Reply:
I'm pretty sure you don't want to call `stat` on arbitrary files, it can throw FileNotFound